apps/sms/views/conversation/js/conversation.js

```diff
@@ -267,48 +267,44 @@ var ConversationView = {
     this.onMessageTypeChange = this.onMessageTypeChange.bind(this);
   },
 
-  onVisibilityChange: function conv_onVisibilityChange(e) {
-    // If we leave the app and are in a thread or compose window
-    // save a message draft if necessary
-    if (document.hidden) {
-      // Auto-save draft if the user has entered anything
-      // in the composer or into To field (for composer panel only).
-      var isAutoSaveRequired = false;
+  onVisibilityChange: function conv_onVisibilityChange() {
+    // If we leave the app and are in a thread or compose window update a
+    // message draft if necessary.
+    var isDraftUpdateNeeded = document.hidden &&
+      (Navigation.isCurrentPanel('composer') ||
+      Navigation.isCurrentPanel('thread'));
 
-      if (Navigation.isCurrentPanel('composer')) {
-        isAutoSaveRequired = !Compose.isEmpty() ||
-          !!ConversationView.recipients.length;
-      } else if (Navigation.isCurrentPanel('thread')) {
-        isAutoSaveRequired = !Compose.isEmpty();
-      }
-
-      if (isAutoSaveRequired) {
-        ConversationView.saveDraft(true /* preserveDraft */);
-      }
+    if (isDraftUpdateNeeded) {
+      this.updateDraft();
     }
   },
 
   /**
-   * Executes when the header 'action' button is tapped.
-   *
-   * @private
-   */
-  onHeaderAction: function conv_onHeaderAction() {
-    this.backOrClose();
-  },
-
-  /**
    * We always go back to the previous pane, unless we're in an activity, then
    * in selected cases we exit the activity.
    *
    * @private
    */
   backOrClose: function conv_backOrClose() {
-    var inActivity = ActivityClient.hasPendingRequest();
```

Comment:
Well, now when we have separate header for information views I don't see much value in 2 separate methods, so I tried to joined ```backOrClose```, ```back``` and ```close``` + move some stuff to ```beforeLeave```.